flang/lib/Optimizer/Transforms/AddDebugInfo.cpp

```diff
@@ -91,6 +169,10 @@ void AddDebugInfoPass::runOnOperation() {
       llvm::dwarf::getLanguage("DW_LANG_Fortran95"), fileAttr, producer,
       isOptimized, debugLevel);
 
+  module.walk([&](fir::GlobalOp globalOp) {
```

Comment:
`mlir::Operation::walk` will visit every descendant operation. I think `fir::GlobalOp` can only be nested directly inside of the `MoudleOp` so it would be much more efficient to have a for loop over the direct descendants of the module.

Reply:
I have changed it to
`for (auto globalOp : module.getOps<fir::GlobalOp>())`

Please let me know if that was not what you had in mind.

Reply:
Yes that's what I meant. Thank you!